test/test_web.py

```diff
@@ -0,0 +1,164 @@
+from datetime import date
+
+from test.testbase import DbIsolatedTest, run_test_as_sync
+
+from db.python.layers.web import (
+    WebLayer,
+    ProjectSummary,
+    NestedParticipant,
+    NestedSample,
+    NestedSequence,
+)
+from db.python.layers.participant import (
+    ParticipantLayer,
+    ParticipantUpsert,
+    ParticipantUpsertBody,
+)
+from db.python.layers.sample import SampleBatchUpsert, SampleLayer
+from db.python.layers.sequence import SampleSequenceLayer, SequenceUpsert
+
+from models.enums.sample import SampleType
+from models.enums.sequencing import SequenceStatus, SequenceType
+
+
+class TestWeb(DbIsolatedTest):
+    """Test web class containing web endpoints"""
+
+    @run_test_as_sync
+    async def setUp(self) -> None:
+        super().setUp()
+        self.webl = WebLayer(self.connection)
+        self.partl = ParticipantLayer(self.connection)
+        self.sampl = SampleLayer(self.connection)
+        self.seql = SampleSequenceLayer(self.connection)
+
+    @run_test_as_sync
+    async def test_project_summary(self):
+        """Test inserting a sample"""
+        result = await self.webl.get_project_summary(token=None)
+
+        # Expect an empty project
+        expected = ProjectSummary(
+            total_samples=0,
+            participants=[],
+            participant_keys=[],
+            sample_keys=[],
+            sequence_keys=[],
+        )
+
+        self.assertEqual(expected, result)
+
+        # Now add a participant with a sample and sequence
+        def data_to_class(data: dict | list) -> dict | list:
```

Comment:
I like this as an idea, can you put the function definition at the bottom of this file to avoid cluttering the function definition?